torch/_logging/_internal.py

```diff
@@ -116,28 +116,110 @@ def clear(self):
 log_state = LogState()
 
 
-# User API for setting log properties
-# ex. format set_logs(LOG_NAME=LEVEL, ARTIFACT_NAME=bool)
-# ex. set_logs(dynamo=logging.DEBUG, graph_code=True)
 def set_logs(
-    dynamo=DEFAULT_LOG_LEVEL,
-    aot=DEFAULT_LOG_LEVEL,
-    inductor=DEFAULT_LOG_LEVEL,
-    bytecode=False,
-    aot_graphs=False,
-    aot_joint_graph=False,
-    graph=False,
-    graph_code=False,
-    guards=False,
-    recompiles=False,
-    output_code=False,
-    schedule=False,
-    modules=None,
+    *,
+    dynamo: int = DEFAULT_LOG_LEVEL,
+    aot: int = DEFAULT_LOG_LEVEL,
+    inductor: int = DEFAULT_LOG_LEVEL,
+    bytecode: bool = False,
+    aot_graphs: bool = False,
+    aot_joint_graph: bool = False,
+    graph: bool = False,
+    graph_code: bool = False,
+    guards: bool = False,
+    recompiles: bool = False,
+    output_code: bool = False,
+    schedule: bool = False,
+    modules: Optional[Dict[str, Union[int, bool]]] = None,
 ):
     """
-    Enable setting the log level of individual components through kwargs.
-    Args are set using the following format:
-        set_logs(<log_name>=<log_level>,...<artifact_name>=<True or False>, ...,modules={"module.qualified.name":<log_level>})
+    Sets the log level for individual components and toggles individual log
+    artifact types.
+
+    .. warning:: This feature is a prototype and may have compatibility
+        breaking changes in the future.
+
+    .. note:: The ``TORCH_LOGS`` environment variable has complete precedence
+        over this function, so if it was set, this function does nothing.
+
+    A component is a set of related features in PyTorch. All of the log
+    messages emitted from a given component has its own log level setting.
+    This allows you to, for instance, silence large groups of log messages that
+    are not relevant to you and increase verbosity of logs for components that
+    are relevant. The expected log level values are:
+
+        * ``logging.CRITICAL``
+        * ``logging.ERROR``
+        * ``logging.WARNING``
+        * ``logging.INFO``
+        * ``logging.DEBUG``
+        * ``logging.NOTSET``
+
+    See documentation for the Python ``logging`` module for more information on
+    log levels: `<https://docs.python.org/3/library/logging.html#logging-levels>`_
+
+    An artifact is a particular type of log message within a component.
+    A component can emit many different kinds of artifacts.
```

Comment:
Does every component belong to a particular artifact? If so, we should explicit this ownership relation, otherwise, we should explain a bit how do the artifacts relate to with the different components.

Reply:
I'm not sure at the moment. I've asked some related questions here: https://github.com/pytorch/pytorch/issues/94788#issuecomment-1513742540

Reply:
Updated this based on @mlazos's answer